cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/templates/etl/batch/ETLBatchTemplate.java

```diff
@@ -100,6 +103,49 @@ public void configureAdapter(String adapterName, ETLBatchConfig etlConfig, Adapt
                                             cronEntry));
   }
 
+  private void validateAdapter(ETLStage source, ETLStage sink, List<ETLStage> transform) throws Exception {
+    String sourceName = source.getName();
+    String sinkName = sink.getName();
+    String sourceClassName = sourceClassMap.get(sourceName);
+    String sinkClassName = sinkClassMap.get(sinkName);
+
+    BatchSource batchSource = (BatchSource) Class.forName(sourceClassName).newInstance();
```

Comment:
these are instantiated in configureSource, configureSink, etc. It would be better to move things around so that they're only created once.